compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1487,7 +1487,7 @@ public void visit(BLangTypeDefinition typeDefinition) {
         }
 
         // check for unresolved fields. This record may be referencing another record
-        if (!this.resolveRecordsUnresolvedDueToFields && typeNodeKind == NodeKind.RECORD_TYPE) {
+        if (hasTypeInclusions && !this.resolveRecordsUnresolvedDueToFields && typeNodeKind == NodeKind.RECORD_TYPE) {
```

Comment:
I am just reverting the change done by the below PR temporarily.
https://github.com/ballerina-platform/ballerina-lang/pull/32095
Need to test its fixed test cases if broken again

Reply:
Fix breaks the test case.